pkg/liqo-controller-manager/namespaceMap-controller/update_namespaceoffloading_status.go

```diff
@@ -0,0 +1,167 @@
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+)
+
+// This function computes the global offloading status according to the number and
+// the type of remote namespaces conditions.
+func computeNamespaceOffloadingStatus(namespaceOffloading *offv1alpha1.NamespaceOffloading, numVirtualNodes int) {
+	if len(namespaceOffloading.Status.RemoteNamespacesConditions) == numVirtualNodes {
+		ready := 0
+		notReady := 0
+
+		for i := range namespaceOffloading.Status.RemoteNamespacesConditions {
+			condition := liqoutils.GetNamespaceOffloadingCondition(
+				namespaceOffloading.Status.RemoteNamespacesConditions[i], offv1alpha1.NamespaceReady)
+			if condition == nil {
+				continue
+			}
+			if condition.Status == corev1.ConditionTrue {
+				ready++
+			} else {
+				notReady++
+			}
+		}
+
+		switch {
+		case ready == 0:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.AllFailedOffloadingPhaseType
+		case notReady == 0:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.ReadyOffloadingPhaseType
+		default:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.SomeFailedOffloadingPhaseType
+		}
+	} else {
+		namespaceOffloading.Status.OffloadingPhase = offv1alpha1.InProgressOffloadingPhaseType
+	}
+}
+
+// If all requested remote namespaces have been created, it is necessary to set the global status of Offloading,
+// otherwise the Offloading is still in progress.
+func (r *NamespaceMapReconciler) setNamespaceOffloadingStatus(namespaceOffloading *offv1alpha1.NamespaceOffloading) error {
+	virtualNodes := &corev1.NodeList{}
+	if err := r.List(context.TODO(), virtualNodes,
+		client.MatchingLabels{liqoconst.TypeLabel: liqoconst.TypeNode}); err != nil {
+		klog.Error(err, " --> Unable to List all virtual nodes")
+		return err
+	}
+
+	// If there are no virtualNodes here I have to reconcile in order to check that also the relative
+	// NamespaceMaps do not exist anymore.
+	if len(virtualNodes.Items) == 0 {
+		err := fmt.Errorf(" No VirtualNode at the moment")
+		klog.Error(err)
+		return err
+	}
+
+	computeNamespaceOffloadingStatus(namespaceOffloading, len(virtualNodes.Items))
+
+	if err := r.Update(context.TODO(), namespaceOffloading); err != nil {
+		klog.Errorf("%s -> unable to update namespaceOffloading for the namespace '%s'",
+			err, namespaceOffloading.Namespace)
+		return err
+	}
+
+	return nil
+}
+
+// This function sets ready condition according to remote namespace status, and checks global offloading status
+// with setNamespaceOffloadingStatus function.
+func (r *NamespaceMapReconciler) setReadyNamespaceCondition(localName, clusterID string, status error) error {
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	if err := r.Get(context.TODO(), types.NamespacedName{
+		Namespace: localName,
+		Name:      liqoconst.DefaultNamespaceOffloadingName,
+	}, namespaceOffloading); err != nil {
+		klog.Errorf("%s -> unable to get namespaceOffloading for the namespace '%s'", err, localName)
+		return err
+	}
```

Comment:
Would prevent code duplication with the `setNotReadyNamespaceCondition` and `removeRemoteNamespaceConditions` functions, to prevent issues during future modifications.

Reply:
you are right, thank you